packages/compass-aggregations/src/components/stage-editor/stage-editor.jsx

```diff
@@ -1,8 +1,7 @@
 import React, { Component } from 'react';
 import PropTypes from 'prop-types';
-import AceEditor from 'react-ace';
 import 'ace-builds';
-import 'ace-builds/webpack-resolver';
+import AceEditor from 'react-ace';
```

Comment:
If you import react-ace before ace-builds it will try and use brace or _something_. Things get weird and hard to reproduce. For hours the app worked but not the unit tests as a result, for example. Just really brittle and depends on import order that way, I suppose.

Reply:
I suspect this is why I had to upgrade mongosh first - mongosh probably got in there first and then it was using brace, not ace-builds.

Reply:
Hmmm, as browser-repl is bundled separately from compass maybe it might be better that we don't bundle ace with it making it a peerDep (or maybe don't bundle browser-repl with webpack at all / provide a special webpack export pointing to source similar to packages in this repo) just to make sure browser-repl is always using the same versions and instances of dependencies that Compass is using. What do you think?

Reply:
Maybe? I don't really understand the webpack implications well enough at all. There's another weird issue there - ace uses worker threads which are separate js files which also get webpacked and I'm not 100% sure that that will work across browser-repl and compass and how to ensure that it does. (see the file-loader bits). We can turn off the workers in config which would greatly simplify things, but I'm not sure if that's a new feature that we don't have in main or if it is old and that would therefore count as a regression... This stuff his hurting my brain.

Reply:
Or exactly which the workers are in play and which packages they come from.. worker_javascript seems to be used _somewhere_, for example.

Reply:
As we mentioned elsewhere - I'll file a ticket for making this a peerDep.

Reply:
https://jira.mongodb.org/browse/COMPASS-5328